controllers/csi/driver/server.go

```diff
@@ -237,6 +246,18 @@ func (svr *CSIDriverServer) NodeUnpublishVolume(_ context.Context, req *csi.Node
 	return &csi.NodeUnpublishVolumeResponse{}, nil
 }
 
+func (svr *CSIDriverServer) loadVolumeMetadata(metadataPath string) (*volumeMetadata, error) {
+	data, err := svr.fs.ReadFile(metadataPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var metadata *volumeMetadata
+	err = json.Unmarshal(data, &metadata)
+
+	return metadata, err
+}
```

Comment:
What should happen if the data in the File is invalid? The returned object would simply contain empty strings.